pkg/controller/issueapi/issue.go

```diff
@@ -290,14 +290,21 @@ func (c *Controller) HandleIssue() http.Handler {
 			IssuingUser:    user,
 			IssuingApp:     authApp,
 			RealmID:        realm.ID,
+			UUID:           request.UUID,
 		}
 
 		code, longCode, uuid, err := codeRequest.Issue(ctx, c.config.GetCollisionRetryCount())
 		if err != nil {
 			logger.Errorw("failed to issue code", "error", err)
-			c.h.RenderJSON(w, http.StatusInternalServerError, api.Errorf("failed to generate otp code, please try again"))
 			blame = observability.BlameServer
 			result = observability.ResultError("FAILED_TO_ISSUE_CODE")
+
+			// GormV1 doesn't have a good way to match db errors
+			if strings.Contains(err.Error(), database.VercodeUUIDUniqueIndex) {
```

Comment:
I think you want to handle this error up one level in `otp.Request.Issue`

otherwise we'll retry the code creation multiple times before popping back up here

and that particular flavor of error is probably not BlamServer?

Reply:
This is also handled in otp.Request.Issue, this is just translating it to be 409 conflict

Reply:
missed that - was looking at the code at head